### DISCUSSION TITLE
integration with starlette-admin

### DISCUSSION THREAD

**Discussion by ridvansg** [fastapi/full-stack-fastapi-template#1294] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched in the documentation/README.
- [X] I already searched in Google "How to do X" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs/README and didn't find an answer.

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```Text
backend/pyproject.toml: starlette-admin = "^0.14.1"

backend/app/main.py: setup_admin(app)

backend/app/ext/admin/admin_config.py:
from fastapi import FastAPI
from starlette_admin.contrib.sqla import Admin, ModelView

def setup_admin(app: FastAPI) -> Admin:
    from app.core.db import engine
    from app.models import Item, User

    # Create admin
    # https://github.com/jowilf/starlette-admin/tree/main
    admin = Admin(engine, base_url="/admin", title="Admin")

    admin.add_view(ModelView(Item))
    admin.add_view(ModelView(User))

    admin.mount_to(app)
```

### Description

Hi Guys, 
congratulations for the great platform that you've build. 
I decided to try it this week and am very happy so far. 

The first thing that I needed was to add something like "flask_admin" to the backend application. 
After checking few options the one that worked best, for me, was the starlette-admin: https://github.com/jowilf/starlette-admin.

I was thinking that maybe you would like to add it to the template?

If yes, do you want a PR?

The code would basically be what is above + some basic tests. 

### Operating System

Linux

### Operating System Details

Ubuntu 21

### Python Version

3.10.12

**Comment by ridvansg** [comment]
Here the patch file attached as .txt. 
To apply it:
- rename it to 0001-integrate-starlette-admin.patch
`git apply --verbose ./0001-integrate-starlette-admin-3.patch`

[0001-integrate-starlette-admin.txt](https://github.com/user-attachments/files/16624736/0001-integrate-starlette-admin.txt)

**Comment by AdamDorwart** [comment]
The project already has [adminer](adminer.org) integrated and deployed alongside. The question should be if starlette-admin should replace adminer (I don't think it's worth it but I've never used starlette-admin).

**Reply by ridvansg** [reply]
Hi, 
starlette-admin is more like a business admin interface, similar to flask_admin and django admin. 
So far starlette-admin works fine for me.

**Reply by silopolis** [reply]
As I see it, the two tools act at very different layers and target potentially completely different audiences.
Having both would be splendid.

**Reply by ridvansg** [reply]
Hi, 

I remember that the patch file above was having some issue with http, https and secure/unsecure references from  script and stylesheet tags, but as there was no much interest I didn't  upload a fixed version. It is fixed on mine, but there are some other things now, which I can't share. 

If there is interest I can fix the patch and provide a new one.
